posts/indu-plp-part3/plp-part3.qmd

```diff
@@ -0,0 +1,147 @@
+
+## Recap of Previous Parts
+
+In **Part 1**, we introduced the motivation and core question driving the pipeline:
+
+> From those diagnosed with hypertension, who goes on to develop diabetes?
+
+We chose this prediction problem because of its clinical relevance and because hypertension and diabetes are both common chronic conditions with strong associations in literature. To handle the data, we used the OMOP CDM format, which ensures that real-world patient data is structured in a consistent and analysis-friendly way.
+
+To extract patient cohorts, we used `OHDSICohortExpressions.jl`, which allowed us to define concept sets and logic for cohort inclusion and exclusion. For example, the hypertension target cohort included patients with a confirmed hypertension diagnosis, and the outcome cohort included those later diagnosed with diabetes.
+
+In **Part 2**, we built a modular Patient-Level Prediction pipeline using the Julia ecosystem:
+
+- **Feature extraction**: involved pulling demographic, condition, drug, and visit data, encoding it as binary presence indicators.
+- **Preprocessing**: included handling missing values, one-hot encoding, normalization, and creating train-test splits.
+- **Model training**: used the `MLJ.jl` framework to train logistic regression, random forest, and XGBoost models.
+- **Evaluation**: involved computing AUC and accuracy for binary classification.
+
+## Reflections on the OHDSI Framework vs Julia Approach
+
+The OHDSI ecosystem, particularly its **Patient Level Prediction** package in R, offers an end-to-end solution that integrates tightly with ATLAS and standardized vocabularies. It handles cohort creation, covariate extraction, modeling, and visualization all in one environment.
+
+<br>
+<center>
+  ![](./ohdsi_plp.png)
+
+  OHDSI Patient Level Prediction Package in R
+</center>
+<br>
+
+In contrast, building this pipeline in Julia provided greater flexibility and control. Each step, from cohort extraction to model evaluation, had to be explicitly implemented. While this required more effort, it allowed deeper insight into how features were generated, how cohorts were structured, and how model performance was affected by upstream choices.
+
+## Model performance
+
+After preprocessing the features and labels, model performance was not as strong as expected. The AUC values for the classifiers were relatively low for the models:
+
+- Logistic Regression (L1-regularized)
+- Random Forest: AUC 
+- XGBoost: AUC
+
+These unexpectedly low model performances prompted a deeper investigation into the root causes. The primary issue was that the data available in our synthetic OMOP CDM was not well-suited to answer the research question: “From those diagnosed with hypertension, who goes on to develop diabetes?” 
```

Comment:
Maybe just add a little bit more about not many patients matched our criteria or the dataset may have been too small, etc.